### PR TITLE
feat: Persist test QTI structure to be used later by DynamicAPI

### PR DESCRIPTION
**Pull request by gabrielfs7**
https://oat-sa.atlassian.net/browse/AUT-3163

# Goal 

Add a new properties for test called `test_qti_strucuture` that stores the QTI Json saved by our QTI SDK and can be queries lated on Dynamic API

# TODO

- [x] Fix unit tests
- [x] Add some migration to add new field

# How to test 

1) Update index

```
php tao/scripts/taoUpdate.php
```

2) Create a test with several test parts and sections 
3) Save the test and check ElasticSearch storage

Example of outcome: 

```json
{
  "took" : 1002,
  "timed_out" : false,
  "_shards" : {
    "total" : 1,
    "successful" : 1,
    "skipped" : 0,
    "failed" : 0
  },
  "hits" : {
    "total" : {
      "value" : 1,
      "relation" : "eq"
    },
    "max_score" : 0.6931471,
    "hits" : [
      {
        "_index" : "gabriel-tests",
        "_type" : "_doc",
        "_id" : "https://test-tao-deploy.docker.localhost/ontologies/tao.rdf#i62068b4d275cd18558b34aec5e26a90db",
        "_score" : 0.6931471,
        "_source" : {
          "label" : [
            "UNIQUE_TEST_ID"
          ],
          "test_qti_structure" : {
            "qti-type" : "assessmentTest",
            "identifier" : "myTestExample",
            "testParts" : [
              {
                "qti-type" : "testPart",
                "identifier" : "testPart-1",
                "navigationMode" : 0,
                "assessmentSections" : [
                  {
                    "qti-type" : "assessmentSection",
                    "sectionParts" : [
                      {
                        "qti-type" : "assessmentItemRef",
                        "href" : "https://test-tao-deploy.docker.localhost/ontologies/tao.rdf#i620fb8a18fbdd85fe9587a3a3fb13d",
                        "identifier" : "item-4"
                      }
                    ],
                    "identifier" : "assessmentSection-1"
                  },
                  {
                    "qti-type" : "assessmentSection",
                    "sectionParts" : [
                      {
                        "qti-type" : "assessmentItemRef",
                        "href" : "https://test-tao-deploy.docker.localhost/ontologies/tao.rdf#i620fb8a18fbdd85fe9587a3a3fb13d",
                        "identifier" : "item-1"
                      }
                    ],
                    "identifier" : "assessmentSection-2"
                  }
                ]
              },
              {
                "qti-type" : "testPart",
                "identifier" : "testPart-2",
                "navigationMode" : 0,
                "assessmentSections" : [
                  {
                    "qti-type" : "assessmentSection",
                    "sectionParts" : [
                      {
                        "qti-type" : "assessmentItemRef",
                        "href" : "https://test-tao-deploy.docker.localhost/ontologies/tao.rdf#i620fb8a18fbdd85fe9587a3a3fb13d",
                        "identifier" : "item-2"
                      }
                    ],
                    "identifier" : "assessmentSection-3"
                  }
                ]
              }
            ]
          },
          "qti_identifier" : "myTestExample"
        }
      }
    ]
  }
}
```

### CODE REVIEW

File: model/Index/Service/SyncResultIndexer.php [hunk 1]
```diff
@@ -84,6 +84,9 @@ private function getSearch(): SearchInterface
 
     private function getIndexerService(): IndexService
     {
-        return $this->getServiceLocator()->get(IndexService::SERVICE_ID);
+        $service = $this->getServiceLocator()->get(IndexService::SERVICE_ID);
```
Comment (gabrielfs7) [inline model/Index/Service/SyncResultIndexer.php:87]: This is actually a bug we have now and some indexation was not working. So it is not directly related to this task, but in order to index the data properly I needed to fix it.

File: migrations/Version202307251344000003_taoAdvancedSearch.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+namespace oat\taoAdvancedSearch\migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use oat\tao\scripts\tools\migrations\AbstractMigration;
+use oat\taoAdvancedSearch\model\SearchEngine\Contract\IndexerInterface;
+use oat\taoAdvancedSearch\scripts\tools\IndexMigration;
+
+final class Version202307251344000003_taoAdvancedSearch extends AbstractMigration
+{
+    private const INDEX_UPDATE_BODY = '{"properties": {"test_qti_structure": {"type": "object", "enabled": false}}}';
```
Comment (hectoras) [inline migrations/Version202307251344000003_taoAdvancedSearch.php:14]: Note for reviewers: This ends up doing a PUT request to add the new properties, see https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-put-mapping.html & https://github.com/oat-sa/extension-tao-advanced-search/blob/28fd95154126a891d30fe16e6d9ef024f7e14ffb/scripts/tools/IndexMigration.php :ok_hand:

File: migrations/Version202307251344000003_taoAdvancedSearch.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
```
Comment (hectoras) [inline migrations/Version202307251344000003_taoAdvancedSearch.php:3]: nitpick: Missing license header :see_no_evil:

**Review by hectoras — APPROVED** [review]
I was not able to test it locally, but the changes look good :muscle: 

- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Target branch is correct
- [x] Change log provided
- [x] Commits are following conventional commits
- [x] New code is covered by tests (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Kitchen links are not exposed in the description

File: migrations/Version202307251344000003_taoAdvancedSearch.php (hunk 3 above)
Comment (gabrielfs7) [reply migrations/Version202307251344000003_taoAdvancedSearch.php:3]: We are not using for any migration file. I am not sure it is needed, but I can add :)

**Review by shpran — APPROVED** [review]
Good job 👍🏻 
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**